app/javascript/pages/admin/Reporting/index.jsx

```diff
@@ -0,0 +1,254 @@
+            stateAndHelpers.setHighlightedIndex(1)
+            break
+          case 'individual':
+            stateAndHelpers.setHighlightedIndex(2)
+            break
+        }
+      }
+
+      setDropDownTempSelectedItem(nextSelectedItem)
+    }
+  }
+
+  const handleDropdownOnSelect = item => {
+    if (item !== 'general' && item !== 'individual'  && item !== 'organized') {
+      setReport(item)
+    }
+  }
+
+  const renderMenu = () => (
+    <Menu >
+      {dropDownTempSelectedItem === 'organized' ? (
+        <>
+          <PreviousItem value="general">{t('volunteer_portal.admin.tab.reporting.dropdown.organized_events')}</PreviousItem>
+          <Separator />
+          <Item value="organizedEventType">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.event_type')}</Item>
+          <Item value="organizedTag">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.tag')}</Item>
+          <Item value="organizedOrganization">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.organization')}</Item>
+          <Item value="organizedTotal">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.total')}</Item>
+        </>
+      ) : dropDownTempSelectedItem === 'individual' ? (
+        <>
+          <PreviousItem value="general">{t('volunteer_portal.admin.tab.reporting.dropdown.individual_events')}</PreviousItem>
+          <Separator />
+          <Item value="individualEvent-type">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.event_type')}</Item>
+          <Item value="individualTag">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.tag')}</Item>
+          <Item value="individualOrganization">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.organization')}</Item>
+          <Item value="individualTotal">{t('volunteer_portal.admin.tab.reporting.dropdown.nested.total')}</Item>
+        </>
+      ) : (
+        <>
+          <Item value="user">{t('volunteer_portal.admin.tab.reporting.dropdown.users')}</Item>
+          <NextItem value="organized" disabled>{t('volunteer_portal.admin.tab.reporting.dropdown.organized_events')}</NextItem>
+          <NextItem value="individual" disabled>{t('volunteer_portal.admin.tab.reporting.dropdown.individual_events')}</NextItem>
+        </>
+      )}
+    </Menu>
+  )
+
+  return (
+    <AdminSection>
+      <div style={{ width: "100%" }}>
+        <ToolbarHeader>
+          <Dropdown
+              isOpen={dropDownIsOpen}
+              onSelect={handleDropdownOnSelect}
+              onStateChange={handleDropdownStateChange}
+            >
+            <Field style={{ width: 200 }}>
+              <FormLabel>{t('volunteer_portal.admin.tab.reporting.select_label')}</FormLabel>
+              <Select>
+                {
+                  report === null ? t('volunteer_portal.admin.tab.reporting.dropdown.none') : reportKeyTitleMap[report]
+                }
+              </Select>
+            </Field>
+            {renderMenu()}
```

Comment:
This is fine, but fyi typically we wouldn't call a local function to render a component like this. Normally the Menu code would be in its own component and we would import it in. The reasoning is that it is easier to see the separations of concerns that way which helps with readability and maintenance.

Reply:
Yes good point!